packages/hooks/README.md

```diff
@@ -1,34 +1,28 @@
 # WP-JS-Hooks
 
-A lightweight & efficient EventManager for JavaScript in WordPress.
-
+A lightweight & efficient event manager.
 
 ### API Usage
-API functions can be called via the global `wp.hooks` like this `wp.hooks.addAction()`, etc.
-
-* `addAction( 'hook', 'vendor/plugin/function', callback, priority )`
-* `addFilter( 'hook', 'vendor/plugin/function', callback, priority )`
-* `removeAction( 'hook', 'vendor/plugin/function' )`
-* `removeFilter( 'hook',  'vendor/plugin/function' )`
-* `removeAllActions( 'hook' )`
-* `removeAllFilters( 'hook' )`
-* `doAction( 'hook', arg1, arg2, moreArgs, finalArg )`
-* `applyFilters( 'hook', content, arg1, arg2, moreArgs, finalArg )`
-* `doingAction( 'hook' )`
-* `doingFilter( 'hook' )`
-* `didAction( 'hook' )`
-* `didFilter( 'hook' )`
-* `hasAction( 'hook' )`
-* `hasFilter( 'hook' )`
-
+Hooks can be added to an object via composition:
+
+`myObject.hooks = new Hooks()`
```

Comment:
In this case, should the `Hooks` function take an optional `context` arg which is then used when running callbacks? For example:

```js
myObject.hooks = new Hooks( myObject );
```

With the parent object then defined, then it would be available for use as `this` context. If not provided, then `this` would be the instance of `hooks` itself.

Reply:
good suggestion, I'll add and see if I can add tests.

Reply:
Should we encourage the use of the context? One of the most controversial JavaScript features. What benefit this would give us?

Also what about `createHooks` instead of `new Hooks()`?

Reply:
The benefit of context I see is that inside your callbacks you could easily refer to the calling object which seems like a common use case, otherwise you would have to bind 'this' to the callback or hoist it to get a handle.

Reply:
>otherwise you would have to bind 'this' to the callback or hoist it to get a handle

An arrow function would work as well, which is not outlandish to target in the near future even without a build step (Vue 3+, for example, is [dropping support for non-ES2015-capable browsers](https://hashnode.com/ama/with-vuejs-team-cj7itlrki03ae62wuv2r2005s#cj7k5nd67003k7qwt1147975x)).